internal/pkg/table/destination.go

```diff
@@ -918,6 +924,25 @@ func compareByRouterID(path1, path2 *Path) (*Path, error) {
 	}
 }
 
+func compareByNeighborAddress(path1, path2 *Path) *Path {
+	// Select the route received from the peer with the lowest peer address as
+	// per RFC 4271 9.1.2.2. g
+	log.WithFields(log.Fields{
+		"Topic": "Table",
+	}).Debug("enter compareByNeighborAddress")
+
+	for i := 0; i < 16; i = i + 4 {
+		a := binary.BigEndian.Uint32(path1.GetSource().Address[i : i+4])
+		b := binary.BigEndian.Uint32(path2.GetSource().Address[i : i+4])
```

Comment:
path1.GetSource().Address is nil if the path is injected locally.

Reply:
thanks for taking a look, updated